code/modules/client/preferences.dm

```diff
@@ -182,14 +182,23 @@ var/const/MAX_SAVE_SLOTS = 8
 /datum/preferences/New(client/C)
 	client=C
 	if(istype(C))
-		if(!IsGuestKey(C.key))
-			var/load_pref = load_preferences_sqlite(C.ckey)
-			if(load_pref)
-				if(load_save_sqlite(C.ckey, C, default_slot))
-					return
-		randomize_appearance_for()
-		real_name = random_name(gender)
-		save_character_sqlite(src, C.ckey, default_slot)
+		if(!speciesinit)
+			spawn(10)
```

Comment:
```
		while(!speciesinit)
			spawn(10)
```
code\modules\client\preferences.dm:187:error: : invalid expression

```
		while(!speciesinit)
			sleep(10)
```
can't join the game until speciesinit returns true